src/App/Fossa/FossaAPIV1.hs

```diff
@@ -452,3 +456,45 @@ uploadContributors apiOpts locator contributors = fossaReq $ do
 
   _ <- req POST (contributorsEndpoint baseUrl) (ReqBodyJson contributors) ignoreResponse opts
   pure ()
+
+----------
+
+newtype IATFingerprintSet = IATFingerprintSet
+  { fingerprintRaw :: Fingerprint Raw
+  }
+
+instance ToJSON IATFingerprintSet where
+  toJSON IATFingerprintSet{..} =
+    object
+      [ "fingerprint_sha_256" .= fingerprintRaw
+      ]
+
+-- | The set of necessary data to describe a user defined binary IAT assertion to the FOSSA API.
+data UserDefinedIATBinaryAssertion = UserDefinedIATBinaryAssertion
+  { userDefinedBinaryAssertionName :: Text
+  , userDefinedBinaryAssertionVersion :: Text
+  , userDefinedBinaryAssertionLicenseIdentifier :: Text
+  , userDefinedBinaryAssertionDescription :: Maybe Text
+  , userDefinedBinaryAssertionUrl :: Maybe Text
+  , userDefinedBinaryAssertionFingerprints :: [IATFingerprintSet]
+  }
```

Comment:
This name and the prefix are a little heavy.  I'll leave it to you to decide the correct name, but here's an idea:
```suggestion
data BinaryAssertion = BinaryAssertion
  { assertionName :: Text
  , assertionVersion :: Text
  , assertionLicenseId :: Text
  , assertionDescription :: Maybe Text
  , assertionUrl :: Maybe Text
  , assertionFingerprints :: [IATFingerprintSet]
  }
```

You should consider removing the IAT acronym everywhere as well.  I'm not sure it adds to the type names and/or function names.

A lot of the details are part of the name, but I'd rather see them as part of the haddocks (doc comments).

Reply:
I think I generally cleaned up the new identifiers and made them fit more of an overall theme as well!